builtins/amp-ad.js

```diff
@@ -98,11 +99,31 @@ export function upgradeImages_(images) {
  * @return {undefined}
  */
 export function installAd(win) {
+
+  /**
+   * @type {boolean} Heuristic boolean as for whether another ad is currently
+   *     loading.
+   */
+  let loadingAdsCount = 0;
+
   class AmpAd extends BaseElement {
 
     /** @override  */
     renderOutsideViewport() {
-      return false;
+      // Before the user has scrolled we only render ads in view. This prevents
```

Comment:
So if the reader has not scroll the page yet, this method will yield `false`. Right? I think a way for quick swiping between articles would be to do some top-level blocking on runtime level for all elements. E.g.  viewer can delay sending `visible` signal to runtime for a little bit. Otherwise, there's always plenty of work to do for runtime to jank up the swiping.